CMakeTests.cmake

```diff
@@ -0,0 +1,228 @@
+    add_h3_test(testCellToChildrenSize src/apps/testapps/testCellToChildrenSize.c)
+    add_h3_test(testH3Index src/apps/testapps/testH3Index.c)
+    add_h3_test(testH3Api src/apps/testapps/testH3Api.c)
+    add_h3_test(testH3SetToLinkedGeo src/apps/testapps/testH3SetToLinkedGeo.c)
+    add_h3_test(testH3SetToVertexGraph src/apps/testapps/testH3SetToVertexGraph.c)
+    add_h3_test(testLinkedGeo src/apps/testapps/testLinkedGeo.c)
+    add_h3_test(testPolygonToCells src/apps/testapps/testPolygonToCells.c)
+    add_h3_test(testPolygonToCellsReported src/apps/testapps/testPolygonToCellsReported.c)
+    add_h3_test(testVertexGraph src/apps/testapps/testVertexGraph.c)
+    add_h3_test(testDirectedEdge src/apps/testapps/testDirectedEdge.c)
+    add_h3_test(testLatLng src/apps/testapps/testLatLng.c)
+    add_h3_test(testBBox src/apps/testapps/testBBox.c)
+    add_h3_test(testVertex src/apps/testapps/testVertex.c)
+    add_h3_test(testPolygon src/apps/testapps/testPolygon.c)
+    add_h3_test(testVec2d src/apps/testapps/testVec2d.c)
+    add_h3_test(testVec3d src/apps/testapps/testVec3d.c)
+    add_h3_test(testH3ToLocalIj src/apps/testapps/testH3ToLocalIj.c)
+    add_h3_test(testGridDistance src/apps/testapps/testGridDistance.c)
+    add_h3_test(testGridPathCells src/apps/testapps/testGridPathCells.c)
+    add_h3_test(testH3CellArea src/apps/testapps/testH3CellArea.c)
+    add_h3_test(testCoordIj src/apps/testapps/testCoordIj.c)
+    add_h3_test(testCoordIjk src/apps/testapps/testCoordIjk.c)
+    add_h3_test(testBaseCells src/apps/testapps/testBaseCells.c)
+    add_h3_test(testPentagonIndexes src/apps/testapps/testPentagonIndexes.c)
+    add_h3_test(testH3Iterators src/apps/testapps/testH3Iterators.c)
+
+    add_h3_test_with_arg(testH3NeighborRotations src/apps/testapps/testH3NeighborRotations.c 0)
+    add_h3_test_with_arg(testH3NeighborRotations src/apps/testapps/testH3NeighborRotations.c 1)
+    add_h3_test_with_arg(testH3NeighborRotations src/apps/testapps/testH3NeighborRotations.c 2)
+
+    # The "Exhaustive" part of the test name is used by the test-fast to exclude these files.
+    # test-fast exists so that Travis CI can run Valgrind on tests without taking a very long time.
+    add_h3_test(testDirectedEdgeExhaustive src/apps/testapps/testDirectedEdgeExhaustive.c)
+    add_h3_test(testVertexExhaustive src/apps/testapps/testVertexExhaustive.c)
+    add_h3_test(testH3ToLocalIjExhaustive src/apps/testapps/testH3ToLocalIjExhaustive.c)
+    add_h3_test(testGridPathCellsExhaustive src/apps/testapps/testGridPathCellsExhaustive.c)
+    add_h3_test(testGridDistanceExhaustive src/apps/testapps/testGridDistanceExhaustive.c)
+    add_h3_test(testH3CellAreaExhaustive src/apps/testapps/testH3CellAreaExhaustive.c)
+
+    add_h3_cli_test(testCliCellToLatLng "cellToLatLng -c 8928342e20fffff" "37.5012466151, -122.5003039349")
+    add_h3_cli_test(testCliLatLngToCell "latLngToCell --lat 20 --lng 123 -r 2" "824b9ffffffffff")
```

Comment:
I still feel like there's a substantive difference between tasks that *run* a test and tasks that *are* tests. This is fine for now, but I don't see how it scales up to a large number of tests - I'd much rather have inputs and outputs defined outside of CMake (in a text file, in yaml, or even as separate lines in a script file).